modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/redirect/button/RedirectButtonDDMFormFieldTemplateContextContributor.java

```diff
@@ -0,0 +1,108 @@
+
+import com.liferay.dynamic.data.mapping.form.field.type.DDMFormFieldTemplateContextContributor;
+import com.liferay.dynamic.data.mapping.form.field.type.constants.DDMFormFieldTypeConstants;
+import com.liferay.dynamic.data.mapping.form.field.type.internal.util.DDMFormFieldTypeUtil;
+import com.liferay.dynamic.data.mapping.model.DDMFormField;
+import com.liferay.dynamic.data.mapping.render.DDMFormFieldRenderingContext;
+import com.liferay.petra.portlet.url.builder.PortletURLBuilder;
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactory;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactoryUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Rodrigo Paulino
+ */
+@Component(
+	immediate = true,
+	property = "ddm.form.field.type.name=" + DDMFormFieldTypeConstants.REDIRECT_BUTTON,
+	service = DDMFormFieldTemplateContextContributor.class
+)
+public class RedirectButtonDDMFormFieldTemplateContextContributor
+	implements DDMFormFieldTemplateContextContributor {
+
+	@Override
+	public Map<String, Object> getParameters(
+		DDMFormField ddmFormField,
+		DDMFormFieldRenderingContext ddmFormFieldRenderingContext) {
+
+		return HashMapBuilder.<String, Object>put(
+			"buttonLabel",
+			DDMFormFieldTypeUtil.getPropertyValue(
+				ddmFormField, ddmFormFieldRenderingContext.getLocale(),
+				"buttonLabel")
+		).put(
+			"message",
+			DDMFormFieldTypeUtil.getPropertyValue(
+				ddmFormField, ddmFormFieldRenderingContext.getLocale(),
+				"message")
+		).put(
+			"redirectURL",
+			() -> {
+				RequestBackedPortletURLFactory requestBackedPortletURLFactory =
+					RequestBackedPortletURLFactoryUtil.create(
+						ddmFormFieldRenderingContext.getHttpServletRequest());
+
+				return PortletURLBuilder.create(
+					requestBackedPortletURLFactory.createActionURL(
+						GetterUtil.getString(
+							(String)ddmFormField.getProperty("portletId")))
+				).setParameters(
+					HashMapBuilder.put(
+						"mvcRenderCommandName",
+						new String[] {
+							GetterUtil.getString(
+								(String)ddmFormField.getProperty(
+									"mvcRenderCommandName"))
+						}
+					).putAll(
+						Stream.of(
+							StringUtil.split(
+								StringUtil.removeChars(
+									GetterUtil.getString(
+										(String)ddmFormField.getProperty(
+											"parameters")),
+									CharPool.CLOSE_BRACKET,
+									CharPool.OPEN_BRACKET))
+						).map(
+							parameter -> parameter.split(StringPool.EQUAL)
+						).collect(
+							Collectors.toMap(
+								keyValue -> keyValue[0],
+								keyValue -> new String[] {keyValue[1]})
+						)
```

Comment:
@rodrigopaulino, what do you think to add this logic to a dedicated method?

I'm feeling that this chaining calls may difficulty a little the comprehension of the code reader, what do you think?

Reply:
Hi, @natocesarrego. Personally, I don't think this is hard to read. I've seen way more complicated calls, but I do believe in the value of formatting and giving a single purpose to a call. But, perhaps we're asking the wrong questions here. Will Brian accept it if we move that part to a different method?

Reply:
Hey @rodrigopaulino, I've already seen he accepts something like below for pulls from Carol:
```
return HashMapBuilder.<String, Object>put(
	"buttonLabel", getButtonLabel())
).put(
```
But, feel free to take a decision regarding this matter. :slightly_smiling_face:

Reply:
Nice, @natocesarrego. Can you share the link to that snippet?

Reply:
Sure! Please check [here](https://github.com/liferay/liferay-portal/blob/575db32a0ec08afa3fd4897140c67de9321d6247/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/checkbox/CheckboxDDMFormFieldTemplateContextContributor.java#L56-L88)

Reply:
Thanks. I see the appeal of extracting a method from the examples you gave, because they cannot actually be inline according to our SF rules. So, since you also gave me the option, I'll leave this as is.

I really appreciate your inputs, @natocesarrego.